.swift-format

```diff
@@ -12,7 +12,7 @@
         "AmbiguousTrailingClosureOverload": false,
         "NoBlockComments": false,
         "OrderedImports": true,
-        "UseLetInEveryBoundCaseVariable": false,
-        "UseSynthesizedInitializer": false
+        "UseLetInEveryBoundCaseVariable": true,
+        "UseSynthesizedInitializer": true
```

Comment:
These are `true` by default, right? So we should be able to remove them from `.swift-format` to make it shorter.

Reply:
That's what I thought as well, but I just noticed that if I remove the line entirely and insert a violation, it isn't flagged.

Oof. I think this is a consequence of the decoding logic; if the `rules` dictionary is present in the JSON, its value is used verbatim. Only if it isn't present at all do we use the default dictionary. Then, when we check if a rule should be applied at a node, if the rule name isn't in the dictionary, it's _always_ false.

We should fix this by always taking the default rule settings and then _merging_ what we read from the JSON into that. That would ensure that the in-memory dictionary always contains all rules. That might reveal some other places in swift-syntax/swift-format where the rules aren't flagging violations, though.

So my thinking is we merge this as-is ("OrderedImports" was already this way) and then fix that for the next release. WDYT?

Reply:
Sounds good to me. But yes, this seems like a swift-format bug to me to not merge the dictionaries that’s worth fixing. If you don’t have time to look at that now, could you file an issue for it?